data/lib/core/creature.lua

```diff
@@ -50,3 +50,38 @@ end
 function Creature.isTile(self)
 	return false
 end
+
+function Creature:setMonsterOutfit(monster, time)
+	local monsterType = MonsterType(monster)
+	if not monsterType then
+		return false
+	end
+
+	if self:getPlayer() and not (getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) or monsterType:isIllusionable()) then
```

Comment:
You can use `self:isPlayer()` here as it's lighter and works the same.

Also wouldn't you want to move `getPlayerFlagValue` to the `Player` metatable before? :P just a suggestion